src/components/Login/Login.component.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import Styled from './Login.styled';
+import { useData } from '../../states/provider';
+import { ACTIONS } from '../../states/reducer';
+
+function ModalLogin({ show }) {
+    const { data, dispatch } = useData();
+    const [userName, setUserName] = useState(undefined);
+    const [password, setPassword] = useState(undefined);
+    const [errorMessage, setErrorMessage] = useState(false);
+
+    function validateUserLogin() {
+        if (data.loginUser) {
+            show(false);
+        } else {
+            setErrorMessage('usuario Invalido');
+        }
+    }
+
+    function handlerLogin() {
+        console.log(data);
+        if (userName && userName.trim().length > 0) {
+            if (password && password.trim().length > 0) {
+                dispatch({
+                    type: ACTIONS.LOGIN,
+                    payload: { user: userName },
+                });
+
+                setTimeout(validateUserLogin, 300);
+            } else {
+                setErrorMessage('Please enter a password');
+            }
+        } else {
+            setErrorMessage('Please enter a username');
+        }
+    }
+
+    return (
+        <div>
```

Comment:
Instead of using an extra div here to suffice to React constraints of wrapping code in one container, you can use React Fragmens so that we don't necessarily add an extra node to the DOM.

You can either say:

```js
<React.Fragment>
    {your code here}
</React.Fragment>
```

Or you can import Fragment from React: 
```js
import { Fragment } from 'react';
...
<Fragment>
    {your code here}
</Fragment>
```